edb/common/markup/renderers/terminal.py

```diff
@@ -641,7 +641,14 @@ def render(markup, *, ensure_newline=True, file=None, renderer=Renderer):
     if file is None:
         file = sys.stdout
 
-    fileno = file.fileno()
+    try:
+        fileno = file.fileno()
+    except OSError:
+        # This is a hack to try to get nice colorized dump output over
+        # a remote-pdb connection. If the output is redirected to
+        # something without fileno, use what ought to be stdout's fileno
+        # to decide on color, etc.
+        fileno = 1
```

Comment:
I'm not so sure about this one, but let's see if it works.

Reply:
Yeah that's about where I am on it.

Reply:
Can you use `sys.stdout.fileno()` instead of hardcoding 1?

Reply:
No, because the whole thing being hacked around here is that sys.stdout has been rebound to a TextIO by the remote debugger or something. Arguably the "right" thing to do here is to drop this hack and just output an unstyled dump, but I really do want the color in this case...